boa/vm/fork.py

```diff
@@ -256,7 +256,7 @@ def try_prefetch_state(self, msg: Message):
                 key = int_to_big_endian(slot)
                 if not self._helper_have_storage(address, slot):
                     account_store._journal_storage[key] = rlp.encode(value)  # type: ignore
-        self.lock_changes()
```

Comment:
@charles-cooper is this correct? `lock_changes` was invalidating the snapshot taken in the boa pytest plugin, crashing the test runner

Reply:
the idea behind using `lock_changes()` was that the items from the RPC should be the "pre-fork state".

we could use a different technique, but i don't think using `commit(snapshot)` is ideal because the fetched RPC state would get rolled back after `anchor()`

Reply:
as discussed elsewhere, the code is now adapted to cache the RPC call